### ISSUE TITLE
Error in Modelica Mechanics

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#373]
**Modified by dietmarw on 1 Jun 2010 08:51 UTC**
There seems to be a problem in `Modelica.Mechanics.MultiBody.Parts.BodyCylinder`
 
The error is in the calculation of `r_CM` which is set to 
```
r_CM[3]=Modelica.Math.Vectors.normalize(r)*length/2 
```
but must be 
```
r_CM[3] = (r + r_shape) / 2;
```

----

**Reported by erik&#x2e;aberg on 1 Jun 2010 08:38 UTC**
One of our customers suspects that there is a bug in Modelica Mechanics.

Best regards,
Erik Åberg

Original message:
 
The problem is in Modelica.Mechanics.MultiBody.Parts.BodyCylinder
 
The error is in the calculation or r_CM which is set to 
r_CM[3]=Modelica.Math.Vectors.normalize(r)\*length/2 

but must be 

r_CM[3] = (r + r_shape) / 2;

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/373

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 1 Jun 2010 08:51 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 26 Jul 2010 19:35 UTC**
Fixed in c1a246c2b5e864994fef16814fb1a3e255d9cd93.

The correct formula is:
  rCM = r_shape + Modelica.Math.Vectors.normalize(lengthDirection)\*length/2

The same error was present in BodyBox and was also corrected.
